### PR TITLE
Pull request for Form validation(client-side) in contact me section

### PR DESCRIPTION
**Pull request by zuhebahmed88091**
Previously added ->

Full desktop version
Add transitions and animations
Add Deployment
When the user clicks (or taps) the hamburger button, the mobile menu appears.
When the user clicks (or taps) the close (X) button, the mobile menu disappears.
When the user clicks (or taps) any of the mobile menu options, the mobile menu disappears.
When the user clicks (or taps) any of the mobile menu options, a correct part of the page is displayed.

For those used the following tools ->

CSS Flexbox
CSS Grid
HTML Semantics
HTML Box Model
HTML Form etc.
CSS Transitions
CSS Animations
GITHUB Page
JavaScript
DOM Manipulation

In this branch add ->

Form validation by using DOM, JavaScript and CSS

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -26,4 +26,44 @@ scrollToLinks.forEach((link) => {
...
+    if (emailValue === '') {
+      setError(email, 'Email is required');
+    }
+  }
+
+  form.addEventListener('submit', (event) => {
+    event.preventDefault();
+    validateFormInputs();
+  });
+});
```
Comment (rotimiazeez) [inline script.js]: Please note that you are yet to implement all the requirements for this milestone. Kindly see the below screenshot for the missing requirement 👍🏽
Screenshot from 2023-05-16 16-38-27

**Review by rotimiazeez — CHANGES_REQUESTED** [review]
# Hi Team 👍🏽,
## Good job so far!
### To Highlight 👇🏽
- No Linter error ✔️ 
- Good PR description 👍🏽 
- Good commit message(s) 😉
- Descriptive README 📖
You still need to work on some issues to prepare your project for the final evaluation, but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR WORDS>


## Suggested changes
_Check the comments under the review._

Please feel free to use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend considering them as they can improve your code._


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag (@rotimiazeez) me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid, unless it is requested otherwise._**

File: script.js [hunk 2]
```diff
@@ -26,4 +26,93 @@ scrollToLinks.forEach((link) => {
...
+//   }
+
+//   function clearError(elm) {
+//     const inputControl = elm.parentElement;
+//     const errorDiv = inputControl.querySelector('.error');
+//     errorDiv.innerText = '';
+//     inputControl.classList.remove('error');
+//   }
+
+//   function validateFormInputs() {
```
Comment (Shisui6) [inline script.js]: Great job implementing the form validation but please make sure to remove all commented out code to keep your codebase clean👍

**Review by Shisui6 — CHANGES_REQUESTED** [review]
Hi @zuhebahmed88091, 

# Requires Changes 🍀
Great work on making the changes requested by a previous reviewer 👍
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet🪲.


 ## Highlights 🎯
 - Email validation done correctly ✅


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions 🪶

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were simply missed by the previous reviewer, and addressing them will really improve your application._

You can also consider the following:

N/A

Cheers and Happy coding!🍻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid, unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Shisui6 — APPROVED** [review]
Hi @zuhebahmed88091, 

# Approved 🪄
Your project is complete! There is nothing else to say other than... it's time to merge it 💎
Congratulations! 🎉 

## Highlights 🎯
- Descriptive PR title and summary ✅
- Follows Github flow ✅
- Good commit messages ✅

## Optional suggestions 🪶

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were simply missed by the previous reviewer, and addressing them will really improve your application._

You can also consider the following:

N/A

Cheers and Happy coding!🍻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**